source/eulerian-path.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include "common.h"
+
+struct edge {
+	int value;
+	list<edge>::iterator it;
+};
+struct EulerPath {
+	struct Edge {
+		int value;
+		list<Edge>::iterator it;
+	};
+	vector<list<Edge>> adj;
+	vi result;
+	int n;
+	void calc() {
+		int odds = 0, start = 0;
+		for(int i = 0; i < n; ++i) {
+			if(adj[i].size() % 2) {
+				++odds;
+				start = i;
+			}
+		}
+		// odds == 2 => eulerian path; odds == 0 => eulerian cycle; else there's no solution
+		assert(odds == 2 || odds == 0); 
```

Comment:
Returning an empty vector instead of crashing seems like a better solution, and there might be cases where we don't know a priori if a solution exists 
```suggestion
		if(odds != 2 && odds != 0) return; // Empty result if there's no path
```